terrawrap/utils/config.py

```diff
@@ -86,6 +89,152 @@ def parse_wrapper_configs(wrapper_config_files: List[str]) -> WrapperConfig:
+        wrapper_config_obj.config = False
+    return wrapper_config_obj
+
+
+def walk_and_graph_directory(starting_dir: str, config_dict) -> Tuple[networkx.DiGraph, List[str]]:
+    """
+    Given a starting directory, walks it and returns all dependency info.
+    :param starting_dir: The starting directory
+    :param config_dict: A dictionary containing wrapper config objects for each seen directory
+    :return: directory_graph: A graph composed of all dependency information for a directory
+    """
+    graph_list = []
+    post_graph_runs = []
+    for root, dirs, _ in os.walk(starting_dir):
+        for name in dirs:
+            dir_path = os.path.join(root, name)
+            has_tf_wrapper = False
+            for file in os.listdir(dir_path):
+                if file.endswith(".tf_wrapper"):
+                    has_tf_wrapper = True
+                    wrapper_config_obj = create_wrapper_config_obj(dir_path)
+                    if not wrapper_config_obj.config:
+                        continue
+                    if wrapper_config_obj.depends_on is None:
+                        post_graph_runs.append(dir_path)
+                        continue
+                    single_config_dependency_graph = networkx.DiGraph()
+                    visited: List[str] = []
+                    graph_wrapper_dependencies(dir_path, config_dict, single_config_dependency_graph, visited)
+                    graph_list.append(single_config_dependency_graph)
+            if not has_tf_wrapper and is_config_directory(dir_path):
+                post_graph_runs.append(dir_path)
+
+    directory_graph = networkx.compose_all(graph_list)
+    return directory_graph, post_graph_runs
+
+
+# pylint: disable=R0912
+def graph_wrapper_dependencies(config_dir: str, config_dict, graph: networkx.DiGraph, visited: List[str]):
+    """
+    Given a directory, recursively finds all other directories it depends on and builds a graph.
+    :param config_dir: The config directory to obtain a dependency graph for
+    :param config_dict: A dictionary containing wrapper config objects for each seen directory
+    :param graph: The graph to add dependency info to. Empty at first, reused in recursion.
+    :param visited: A list of visited nodes. Empty at first, reused in recursion
+    """
+    if config_dir in visited:
+        return
+    visited.append(config_dir)
+
+    if config_dict.get(config_dir):  # add to dictionary so we only read the file once
+        wrapper_config_obj = config_dict[config_dir].get("wrapper_config")
+    else:
+        wrapper_config_obj = create_wrapper_config_obj(config_dir)
+        config_dict[config_dir] = {
+            "wrapper_config": wrapper_config_obj
+        }
+
+    if wrapper_config_obj.config:
+        graph.add_node(config_dir)
+
+    tf_dependencies = wrapper_config_obj.depends_on
+
+    if tf_dependencies is None:
+        print("Cannot list a dependency without tf_wrapper dependency configuration:", config_dir)
```

Comment:
This means the user listed a dependency in their wrapper file such as `- foo/bar` and that file does not have dependency information of its own.